examples/event_handling/EventHandling.cpp

```diff
@@ -0,0 +1,224 @@
+                m_window.handleEvents(
+                    [&](auto&& event)
+                    {
+                        // Remove reference and cv-qualifiers
+                        using T = std::decay_t<decltype(event)>;
+
+                        if constexpr (std::is_same_v<T, sf::Event::Closed>)
+                        {
+                            m_window.close();
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::KeyPressed>)
+                        {
+                            m_log.emplace_back("2: Key Pressed: " + sf::Keyboard::getDescription(event.scancode));
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::MouseMoved>)
+                        {
+                            m_log.emplace_back("2: Mouse Moved: " + vec2ToString(event.position));
+                        }
+                        if constexpr (std::is_same_v<T, sf::Event::MouseButtonPressed>)
+                        {
+                            m_log.emplace_back("2: Mouse Pressed");
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::TouchBegan>)
+                        {
+                            m_log.emplace_back("2: Touch Began: " + vec2ToString(event.position));
+                        }
+                        else
+                        {
+                            // All unhandled events will end up here
+                            // m_log.emplace_back("2: Other Event");
+                        }
+                    });
+            }
+            else if (handlerType == 3)
+            {
+                // Forward to other callable
+                // In this case we forward it to our handle member functions
+                // we defined for the concrete event types we want to handle
+                // When choosing this method a default "catch-all" handler must
+                // be available for unhandled events to be forwarded to
+                // If you don't want to provide an empty "catch-all" handler
+                // you will have to make sure (e.g. via if constexpr) that this
+                // lambda doesn't attempt to call a member function that doesn't exist
+                m_window.handleEvents([this](const auto& event) { handle(event); });
+            }
+
+            // Switch to next form of event handling
+            handlerType = (handlerType + 1) % 4;
+
+            while (m_log.size() > 24u)
+                m_log.erase(m_log.begin());
+
+            // Draw the contents of the log to the window
+            m_window.clear();
+
+            for (std::size_t i = 0; i < m_log.size(); ++i)
+            {
+                m_logText.setPosition({50.f, static_cast<float>(i * 20) + 50.f});
+                m_logText.setString(m_log[i]);
+                m_window.draw(m_logText);
+            }
```

Comment:
I would have opted to only update the `sf::Text`, if the log actually changed.

Reply:
That would make the example longer and more complex than necessary. Performance isn't really a concern here.